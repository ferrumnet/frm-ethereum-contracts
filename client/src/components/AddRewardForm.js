import React from 'react';

const AddRewardForm = ({ handleChange, handleSubmit, state }) => (
  <div className="container col-sm-4 col-offset-3">
      <div className="card">
    <div className="card-body">
    <form onSubmit={handleSubmit}>
      <div className="form-group">
        <label >Reward Amount</label>
        <input
          type="number"
          className="form-control"
          name="amount"
          aria-describedby="emailHelp"
          placeholder="Enter email"
          onChange={handleChange}
          value={state.amount}
        />
      </div>
      <div className="form-group">
        <label >Withdraw Amount</label>
        <input
          type="number"
          className="form-control"
          name="withdrawable"
          placeholder="Password"
          onChange={handleChange}
          value={state.withdrawable}
        />
      </div>
      <button type="submit" className="btn btn-primary">
        Add Reward
      </button>
    </form>
    </div> 
  </div>
  </div>
);


export default AddRewardForm;
