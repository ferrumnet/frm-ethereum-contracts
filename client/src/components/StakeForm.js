import React from 'react';

const StakeForm = ({handleChange, handleSubmit, amount}) => (
    <div className="container col-sm-4 col-offset-3">
    <div className="card">
    <div className="card-body">
    <form onSubmit={handleSubmit}>
      <div className="form-group">
        <label>Stake Amount</label>
        <input
          name="amount"
          type="number"
          className="form-control"
          onChange={handleChange}
          value={amount}
        />
      </div>
      <button type="submit" className="btn btn-primary">
        Stake
      </button>
    </form>
  </div>
  </div>
  </div>

);


export default StakeForm;