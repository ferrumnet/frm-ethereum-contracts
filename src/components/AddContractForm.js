import React from "react";
const AddContractForm = ({ handleChange, address, handleSubmit }) => (
  <div className="card">
    <div className="card-body">
      <form onSubmit={handleSubmit}>
        <div className="form-group">
          <label>Add Contract Address</label>
          <input
            name="address"
            autoComplete="off"
            type="text"
            id="address-input"
            className="form-control"
            onChange={handleChange}
            value={address}
          />
        </div>
        <button type="submit" className="btn btn-primary">
          Submit
        </button>
      </form>
    </div>
  </div>
);
export default AddContractForm;
