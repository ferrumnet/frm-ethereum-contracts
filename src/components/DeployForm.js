import React from "react";

const DeployForm = ({
  stakingCap,
  stakingStart,
  stakingEnd,
  withdrawStart,
  withdrawEnd,
  handleChange,
  handleSubmit,
  loading
}) => (
  <div className="col-sm-8 offset-sm-2 top-padding" >
    <div className="card">
      <div className="card-body">
        <form onSubmit={handleSubmit}>
          <div className="form-group row col-sm-12">
            <label className="col-sm-4">Staking Cap</label>
            <input
              type="number"
              id="stakingCap"
              name="stakingCap"
              onChange={handleChange}
              value={stakingCap}
              className="col-sm-7 form-control"
            />
          </div>
          <div className="form-group row col-sm-12">
            <label className="col-sm-4">Staking Start Date</label>
            <input
              type="date"
              id="stakingStart"
              name="stakingStart"
              onChange={handleChange}
              value={stakingStart}
              className="col-sm-7 form-control"
              required
            />
          </div>
          <div className="form-group row col-sm-12">
            <label className="col-sm-4">Staking End Date</label>
            <input
              type="date"
              id="stakingEnd"
              name="stakingEnd"
              onChange={handleChange}
              value={stakingEnd}
              className="col-sm-7 form-control"
              required
            />
          </div>
          <div className="form-group row col-sm-12">
            <label className="col-sm-4">withdraw Start Date</label>
            <input
              type="date"
              id="withdrawStart"
              name="withdrawStart"
              onChange={handleChange}
              value={withdrawStart}
              className="col-sm-7 form-control"
              required
            />
          </div>
          <div className="form-group row col-sm-12">
            <label className="col-sm-4">withdraw End Date</label>
            <input
              type="date"
              id="withdrawEnd"
              name="withdrawEnd"
              onChange={handleChange}
              value={withdrawEnd}
              className="col-sm-7 form-control"
              required
            />
          </div>
          <div className="form-group text-center">
            <button
              type="submit"
              className="btn btn-md col-sm-4  form-control btn-primary"
              disabled={loading}
            >
              {loading ? (
                <>
                  <span
                    className="spinner-border spinner-border-sm"
                    role="status"
                    aria-hidden="true"
                  ></span>{" "}
                  Loading...
                </>
              ) : (
                "Deploy Contract"
              )}
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
);

export default DeployForm;
