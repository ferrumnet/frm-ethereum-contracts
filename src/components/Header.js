/* eslint-disable jsx-a11y/anchor-is-valid */
import React from "react";
import { NavLink, withRouter } from "react-router-dom";

const Header = ({
  history: {
    location: { pathname }
  }
}) => {
  const isAdmin = pathname.match(/admin/g) ? true : false;

  const activeStyle = { color: "#FFFFFF" };
  return (
    <header>
      <nav className="navbar  navbar-expand-lg navbar-dark py-4 bg-dark">
        <NavLink className="navbar-brand" to="/">
          Ferrum Contracts
        </NavLink>
        <button
          className="navbar-toggler"
          type="button"
          data-toggle="collapse"
          data-target="#navbarText"
          aria-controls="navbarText"
          aria-expanded="false"
          aria-label="Toggle navigation"
        >
          <span className="navbar-toggler-icon" />
        </button>
        <div className="collapse navbar-collapse" id="navbarText">
          <ul className="navbar-nav mr-auto ">
            {isAdmin && (
                <li className="nav-item">
                  <NavLink
                    className="nav-link"
                    exact
                    to="/admin/deploy"
                    activeStyle={activeStyle}
                  >
                    Deploy
                  </NavLink>
                </li>

            )}
          </ul>
          <ul className="navbar-nav ml-auto">
            <li className="nav-item dropdown">
              <a
                className="nav-link dropdown-toggle"
                id="navbarDropdownMenuLink-4"
                data-toggle="dropdown"
                aria-haspopup="true"
                aria-expanded="false"
              >
                <i className="fas fa-user"></i> Account
              </a>
              <div
                className="dropdown-menu dropdown-menu-right p-2"
                aria-labelledby="navbarDropdownMenuLink-4"
              >
                <NavLink
                  className="dropdown-item"
                  exact
                  to="/admin/deploy"
                  activeStyle={activeStyle}
                >
                  Admin
                </NavLink>
                <NavLink
                  className="dropdown-item"
                  exact
                  to="/"
                  activeStyle={activeStyle}
                >
                  Customer
                </NavLink>
              </div>
            </li>
          </ul>
        </div>
      </nav>
    </header>
  );
};

export default withRouter(Header);
