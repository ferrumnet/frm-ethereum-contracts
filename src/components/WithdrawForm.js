import React from "react";
import WithdrawForm from "./common/MultiStepForm";
import MultiStepCard from "./common/MultiStepCard";

export const WithdrawStepper = ({
  amount,
  step,
  handleSubmit,
  handleChange,
  prevStep,
  validateWithdraw,
  disable
}) => {
  switch (step) {
    case 1:
      return (
        <div className="min-component-height top-padding">
          <WithdrawForm
            nextStep={validateWithdraw}
            handleChange={handleChange}
            amount={amount}
            title="Withdraw"
            disable={disable}
          />
        </div>
      );
    case 2:
      return (
        <div className="min-component-height top-padding">
          <MultiStepCard
            nextStep={handleSubmit}
            amount={amount}
            cancel={prevStep}
            buttonName="Submit"
            message="Do you want to proceed with the withdraw of"
            title="Confirm Transaction"
            disable={disable}
          />
        </div>
      );
    default:
      return (
        <div className="min-component-height top-padding">
          <WithdrawForm
            validateWithdraw={validateWithdraw}
            handleChange={handleChange}
            amount={amount}
            disable={disable}
          />
        </div>
      );
  }
};
