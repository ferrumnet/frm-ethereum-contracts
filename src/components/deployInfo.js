import React from "react";

const DeployInfo = ({
  stakingCap,
  stakingStart,
  stakingEnd,
  withdrawStart,
  withdrawEnd,
  loading
}) => (
  <div className="col-sm-4 top-padding">
    {loading ? (
      "Loading ...."
    ) : (
      <div className="card">
        <div className="card-header">Deployment Information</div>
        <ul className="list-group list-group-flush">
          <li className="list-group-item">
            <strong>Staking Cap: </strong> {stakingCap}
          </li>
          <li className="list-group-item">
            <strong>Staking Start Date: </strong> {stakingStart}
          </li>
          <li className="list-group-item">
            <strong>Staking End Date:</strong> {stakingEnd}
          </li>
          <li className="list-group-item">
            <strong>withdraw Start Date:</strong> {withdrawStart}
          </li>
          <li className="list-group-item">
            <strong>withdraw End Date:</strong> {withdrawEnd}
          </li>
        </ul>
      </div>
    )}
  </div>
);

export default DeployInfo;
