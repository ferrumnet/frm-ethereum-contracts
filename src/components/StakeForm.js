import React from "react";
import StakeForm from "./common/MultiStepForm";
import MultiStepCard from "./common/MultiStepCard";

export const StakeStepper = ({
  amount,
  step,
  handleSubmit,
  handleChange,
  cancel,
  validateStake,
  authorizeStake,
  disable
}) => {
  switch (step) {
    case 1:
      return (
        <div className="min-component-height top-padding">
          <StakeForm
            nextStep={validateStake}
            handleChange={handleChange}
            amount={amount}
            disable={disable}
            title="Stake"
          />
        </div>
      );
    case 2:
      return (
        <div className="min-component-height top-padding">
          <MultiStepCard
            cancel={cancel}
            amount={amount}
            nextStep={authorizeStake}
            buttonName="Approve"
            message="Do you want to approve stake of"
            title="Approve Stake"
            disable={disable}
          />
        </div>
      );
    case 3:
      return (
        <div className="min-component-height top-padding">
          <MultiStepCard
            nextStep={handleSubmit}
            amount={amount}
            cancel={cancel}
            buttonName="Submit"
            message="Do you want add a stake of"
            title="Confirm Stake"
            disable={disable}
          />
        </div>
      );
    default:
      return (
        <div className="min-component-height top-padding">
          <StakeForm
            nextStep={validateStake}
            handleChange={handleChange}
            amount={amount}
            title="Stake"
            disable={disable}
          />
        </div>
      );
  }
};
