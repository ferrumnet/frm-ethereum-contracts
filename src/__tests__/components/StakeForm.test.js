import { StakeStepper } from "../../components/StakeForm";
import { shallow } from "enzyme";
import React from "react";

describe("Test StakeForm Component", () => {
  test("It should match the snapshot of StakeStepper when step is 1", () => {
    const props = {
      amount: 1000,
      step: 1,
      handleSubmit: jest.fn(),
      handleChange: jest.fn(),
      cancel: jest.fn(),
      validateStake: jest.fn(),
      authorizeStake: jest.fn()
    };
    const wrapper = shallow(<StakeStepper {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("It should match the snapshot of StakeStepper when step is 2", () => {
    const props = {
      amount: 1000,
      step: 2,
      handleSubmit: jest.fn(),
      handleChange: jest.fn(),
      cancel: jest.fn(),
      validateStake: jest.fn(),
      authorizeStake: jest.fn()
    };
    const wrapper = shallow(<StakeStepper {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("It should match the snapshot of StakeStepper when step is 3", () => {
    const props = {
      amount: 1000,
      step: 3,
      handleSubmit: jest.fn(),
      handleChange: jest.fn(),
      cancel: jest.fn(),
      validateStake: jest.fn(),
      authorizeStake: jest.fn()
    };
    const wrapper = shallow(<StakeStepper {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("It should match the snapshot of StakeStepper on default", () => {
    const props = {
      amount: 1000,
      step: null,
      handleSubmit: jest.fn(),
      handleChange: jest.fn(),
      cancel: jest.fn(),
      validateStake: jest.fn(),
      authorizeStake: jest.fn()
    };
    const wrapper = shallow(<StakeStepper {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
