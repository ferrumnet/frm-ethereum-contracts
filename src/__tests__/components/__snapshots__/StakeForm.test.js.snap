// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test StakeForm Component It should match the snapshot of StakeStepper on default 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StakeStepper
    amount={1000}
    authorizeStake={[MockFunction]}
    cancel={[MockFunction]}
    handleChange={[MockFunction]}
    handleSubmit={[MockFunction]}
    step={null}
    validateStake={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <MultiStepForm
        amount={1000}
        handleChange={[MockFunction]}
        nextStep={[MockFunction]}
        title="Stake"
      />,
      "className": "min-component-height top-padding",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "amount": 1000,
        "handleChange": [MockFunction],
        "nextStep": [MockFunction],
        "title": "Stake",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <MultiStepForm
          amount={1000}
          handleChange={[MockFunction]}
          nextStep={[MockFunction]}
          title="Stake"
        />,
        "className": "min-component-height top-padding",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "amount": 1000,
          "handleChange": [MockFunction],
          "nextStep": [MockFunction],
          "title": "Stake",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;

exports[`Test StakeForm Component It should match the snapshot of StakeStepper when step is 1 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StakeStepper
    amount={1000}
    authorizeStake={[MockFunction]}
    cancel={[MockFunction]}
    handleChange={[MockFunction]}
    handleSubmit={[MockFunction]}
    step={1}
    validateStake={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <MultiStepForm
        amount={1000}
        handleChange={[MockFunction]}
        nextStep={[MockFunction]}
        title="Stake"
      />,
      "className": "min-component-height top-padding",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "amount": 1000,
        "handleChange": [MockFunction],
        "nextStep": [MockFunction],
        "title": "Stake",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <MultiStepForm
          amount={1000}
          handleChange={[MockFunction]}
          nextStep={[MockFunction]}
          title="Stake"
        />,
        "className": "min-component-height top-padding",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "amount": 1000,
          "handleChange": [MockFunction],
          "nextStep": [MockFunction],
          "title": "Stake",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;

exports[`Test StakeForm Component It should match the snapshot of StakeStepper when step is 2 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StakeStepper
    amount={1000}
    authorizeStake={[MockFunction]}
    cancel={[MockFunction]}
    handleChange={[MockFunction]}
    handleSubmit={[MockFunction]}
    step={2}
    validateStake={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <MultiStepCard
        amount={1000}
        buttonName="Approve"
        cancel={[MockFunction]}
        message="Do you want to approve stake of"
        nextStep={[MockFunction]}
        title="Approve Stake"
      />,
      "className": "min-component-height top-padding",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "amount": 1000,
        "buttonName": "Approve",
        "cancel": [MockFunction],
        "message": "Do you want to approve stake of",
        "nextStep": [MockFunction],
        "title": "Approve Stake",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <MultiStepCard
          amount={1000}
          buttonName="Approve"
          cancel={[MockFunction]}
          message="Do you want to approve stake of"
          nextStep={[MockFunction]}
          title="Approve Stake"
        />,
        "className": "min-component-height top-padding",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "amount": 1000,
          "buttonName": "Approve",
          "cancel": [MockFunction],
          "message": "Do you want to approve stake of",
          "nextStep": [MockFunction],
          "title": "Approve Stake",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;

exports[`Test StakeForm Component It should match the snapshot of StakeStepper when step is 3 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StakeStepper
    amount={1000}
    authorizeStake={[MockFunction]}
    cancel={[MockFunction]}
    handleChange={[MockFunction]}
    handleSubmit={[MockFunction]}
    step={3}
    validateStake={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <MultiStepCard
        amount={1000}
        buttonName="Submit"
        cancel={[MockFunction]}
        message="Do you want add a stake of"
        nextStep={[MockFunction]}
        title="Confirm Stake"
      />,
      "className": "min-component-height top-padding",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "amount": 1000,
        "buttonName": "Submit",
        "cancel": [MockFunction],
        "message": "Do you want add a stake of",
        "nextStep": [MockFunction],
        "title": "Confirm Stake",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <MultiStepCard
          amount={1000}
          buttonName="Submit"
          cancel={[MockFunction]}
          message="Do you want add a stake of"
          nextStep={[MockFunction]}
          title="Confirm Stake"
        />,
        "className": "min-component-height top-padding",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "amount": 1000,
          "buttonName": "Submit",
          "cancel": [MockFunction],
          "message": "Do you want add a stake of",
          "nextStep": [MockFunction],
          "title": "Confirm Stake",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
